### PR TITLE
makecert: Fix double free crashes

### PR DESCRIPTION
**Pull request by mfleisz**

### CODE REVIEW

File: winpr/tools/makecert/makecert.c [hunk 1]
```diff
@@ -326,7 +326,7 @@ int command_line_pre_filter(MAKECERT_CONTEXT* context, int index, int argc, LPCS
 	if (index == (argc - 1))
 	{
 		if (argv[index][0] != '-')
-			context->output_file = (char*) argv[index];
+			context->output_file = _strdup(argv[index]);
```
Comment (bmiklautz) [inline winpr/tools/makecert/makecert.c]: Unchecked strdup.

File: winpr/tools/makecert/makecert.c [hunk 2]
```diff
@@ -728,7 +728,7 @@ int makecert_context_output_private_key_file(MAKECERT_CONTEXT* context, char* pa
 		return 1;
 
 	if (!context->output_file)
-		context->output_file = context->default_name;
+		context->output_file = _strdup(context->default_name);
```
Comment (bmiklautz) [inline winpr/tools/makecert/makecert.c]: Unchecked strdup.

**Comment by mfleisz** [comment]
makecert used to crash with following command line: winpr-makecert.exe -silent -rdp Default

**Comment by bmiklautz** [comment]
@freerdp-bot test this.

File: winpr/tools/makecert/makecert.c [hunk 3]
```diff
@@ -728,7 +732,11 @@ int makecert_context_output_private_key_file(MAKECERT_CONTEXT* context, char* pa
 		return 1;
 
 	if (!context->output_file)
-		context->output_file = context->default_name;
+	{
+		context->output_file = _strdup(context->default_name);
+		if (context->output_file)
```
Comment (bmiklautz) [inline winpr/tools/makecert/makecert.c]: I think there is a ! missing.

**Comment by akallabeth** [comment]
+1

**Comment by bmiklautz** [comment]
+1

**Pull request merged**